src/main/java/world/bentobox/bentobox/api/flags/clicklisteners/CycleClick.java

```diff
@@ -66,30 +67,46 @@ public boolean onClick(Panel panel, User user, ClickType click, int slot) {
         }
         // Left clicking increases the rank required
         // Right clicking decreases the rank required
+        // Shift Left Click toggles player visibility
         // Get the user's island
         island = plugin.getIslands().getIslandAt(user.getLocation()).orElse(plugin.getIslands().getIsland(user.getWorld(), user.getUniqueId()));
         if (island != null && (user.isOp() || user.getUniqueId().equals(island.getOwner()))) {
             changeOccurred = true;
             RanksManager rm = plugin.getRanksManager();
             plugin.getFlagsManager().getFlag(id).ifPresent(flag -> {
+
+                // Flag visibility
+                boolean invisible = false;
+                // Rank
                 int currentRank = island.getFlag(flag);
                 if (click.equals(ClickType.LEFT)) {
                     if (currentRank >= maxRank) {
                         island.setFlag(flag, minRank);
                     } else {
                         island.setFlag(flag, rm.getRankUpValue(currentRank));
                     }
-                    user.getPlayer().playSound(user.getLocation(), Sound.BLOCK_STONE_BUTTON_CLICK_ON, 1F, 1F);
+                    user.getPlayer().playSound(user.getLocation(), Sound.BLOCK_STONE_BUTTON_CLICK_OFF, 1F, 1F);
                 } else if (click.equals(ClickType.RIGHT)) {
                     if (currentRank <= minRank) {
                         island.setFlag(flag, maxRank);
                     } else {
                         island.setFlag(flag, rm.getRankDownValue(currentRank));
                     }
                     user.getPlayer().playSound(user.getLocation(), Sound.BLOCK_STONE_BUTTON_CLICK_ON, 1F, 1F);
+                } else if (click.equals(ClickType.SHIFT_LEFT) && user.isOp()) {
+                    if (!plugin.getIWM().getVisibleSettings(user.getWorld()).contains(flag.getID())) {
```

Comment:
Didn't you say it was meant to be "hidden" there?

Reply:
Yes, I'll rename this method.